mmv1/third_party/tgc_next/pkg/tfplan2cai/resolvers/advanced_resolver.go

```diff
@@ -0,0 +1,139 @@
+package resolvers
+
+import (
+	"slices"
+	"sort"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"github.com/GoogleCloudPlatform/terraform-google-conversion/v7/pkg/tfplan2cai/models"
+	"github.com/GoogleCloudPlatform/terraform-google-conversion/v7/pkg/tfplan2cai/tfplan"
+
+	provider "github.com/GoogleCloudPlatform/terraform-google-conversion/v7/pkg/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// List of keyword to filter out in order to find the iam resource parent
+var filterList = []string{"etag", "id", "role", "members", "condition", "member"}
+
+type AdvancedPreResolver struct {
+	schema *schema.Provider
+
+	// For logging error / status information that doesn't warrant an outright failure
+	errorLogger *zap.Logger
+}
+
+func NewAdvancedResolver(errorLogger *zap.Logger) *AdvancedPreResolver {
+	return &AdvancedPreResolver{
+		schema:      provider.Provider(),
+		errorLogger: errorLogger,
+	}
+}
+
+func (r *AdvancedPreResolver) Resolve(jsonPlan []byte, resourceDataMap map[string][]*models.FakeResourceDataWithMeta) map[string][]*models.FakeResourceDataWithMeta {
+	// ReadChanges
+	planChanges, err := tfplan.ReadResourceChanges(jsonPlan)
+	if err != nil {
+		return resourceDataMap
+	}
+
+	// Read elements from the resouce config
+	resourceConfig, err := tfplan.ReadResourceConfigurations(jsonPlan)
+	if err != nil {
+		return resourceDataMap
+	}
+
+	// Key is iam resource, values are ids
+	idToAddressMap := make(map[string][]string)
+
+	for _, rc := range planChanges {
+		// Silently skip non-google resources
+		if !strings.HasPrefix(rc.Type, "google_") {
+			continue
+		}
+		// Handle iam resources, build an id for each of them and group them together
```

Comment:
We can add a check if the resource address is in resourceDataMap. If not, then continue to the next iteration in the for loop. With this check, I think we don't need the transformation from line 114 - 128.